code/include/playfab_c/PlayFabClientApi_c.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include "httpClient/pal.h"
+#include "httpClient/async.h"
+#include "PlayFabClientDataModels_c.h"
+
+/// <summary>
+/// Initialize PlayFab global state
+/// </summary>
+HRESULT PlayFabInitialize(
+    _In_ const char* titleId
+) noexcept;
+
+/// <summary>
+/// Cleanup PlayFab global state. Shouldn't be called while async APIs are still running.
+/// </summary>
+HRESULT PlayFabCleanupAsync(
+    _In_ XAsyncBlock* async
+) noexcept;
+
+/// <summary>
+/// Get the result handle for an asynchronous API
+/// </summary>
+HRESULT PlayFabGetResultHandle(
+    _In_ XAsyncBlock* async,
+    _Out_ PlayFabResultHandle* resultHandle
+) noexcept;
+
+/// <summary>
+/// Duplicate result.
+/// </summary>
+HRESULT PlayFabResultDuplicateHandle(
+    _In_ PlayFabResultHandle resultHandle,
+    _Out_ PlayFabResultHandle* duplicatedHandle
+) noexcept;
+
+/// <summary>
+/// Close result. Invalidates all pointers associated with the result handle.
+/// </summary>
+void PlayFabResultCloseHandle(
+    _In_ PlayFabResultHandle resultHandle
+) noexcept;
+
+/// <summary>
+/// Login with custom ID
+/// </summary>
+HRESULT PlayFabLoginWithCustomIDAsync(
+    _In_ const PlayFabLoginWithCustomIDRequest* request,
+    _In_ XAsyncBlock* async
+) noexcept;
+
+/// <summary>
+/// Retrieves PlayFabUserHandle from a login result. User handle is needed to make future API calls
+/// </summary>
+HRESULT PlayFabLoginWithCustomIDResultGetUserHandle(
+    _In_ PlayFabResultHandle resultHandle,
+    _Out_ PlayFabUserHandle* userHandle
+) noexcept;
+
+/// <summary>
+/// Retrieves playFabId from a PlayFabLoginWithCustomIDAsync result.
+/// </summary>
+HRESULT PlayFabLoginWithCustomIdResultGetPlayFabId(
+    _In_ PlayFabResultHandle resultHandle,
+    _Out_ const char** playFabId
+) noexcept;
+
+/// <summary>
+/// Get Player Profile
+/// </summary>
+HRESULT PlayFabGetPlayerProfileAsync(
+    _In_ PlayFabUserHandle user,
+    _In_ const PlayFabGetPlayerProfileRequest* request,
+    _In_ XAsyncBlock* async
+) noexcept;
+
+/// <summary>
+/// Retrieves the Player Profile model from a PlayFabGetPlayerProfileAsync result
+/// </summary>
+HRESULT PlayFabGetPlayerProfileResultGetProfile(
+    _In_ PlayFabResultHandle resultHandle,
+    _Out_ PlayFabPlayerProfileModel** profile
+) noexcept;

```

Comment:
same comment here.  could be:

HRESULT PlayFabGetPlayerProfileResult(
    _In_ XAsyncBlock* async,
    _Out_ PlayFabPlayerProfileModel** profile
) noexcept;

Reply:
Another consideration that I wanted to mention: from a service perspective the "profile" object is a field in the result object.  In this case it happens to be the only field of the result, but for other APIs the result has multiple fields (see PlayFabClientDataModels.h).  It feels a little weird to get the result handle and then call a second API to get the single sub-object from the result, but I want to avoid special cases and have a pattern that is reusable across all APIs, particularly since this will all be auto generated soon.

---
In reply to: [590789416](https://github.com/jplafonta/XPlatCppSdk/pull/2#discussion_r590789416) [](ancestors = 590789416,590782393)

Reply:
Design slightly change.  Single API returns both handle and object model pointer

---
In reply to: [591719925](https://github.com/jplafonta/XPlatCppSdk/pull/2#discussion_r591719925) [](ancestors = 591719925,590789416,590782393)